src/libraries/Microsoft.PowerFx.Interpreter/Functions/CollectFunction.cs

```diff
@@ -0,0 +1,224 @@
+        {
+            Contracts.AssertValue(args);
+            Contracts.AssertAllValues(args);
+            Contracts.AssertValue(argTypes);
+            Contracts.Assert(args.Length == argTypes.Length);
+            Contracts.AssertValue(errors);
+            Contracts.Assert(MinArity <= args.Length && args.Length <= MaxArity);
+
+            var fValid = base.CheckInvocation(binding, args, argTypes, errors, out returnType, out nodeToCoercedTypeMap);
+            Contracts.Assert(returnType.IsTable);
+
+            DType dataSourceType = argTypes[0];
+
+            // Need a collection for the 1st arg
+            DType collectionType = argTypes[0];
+            if (!collectionType.IsTable)
+            {
+                errors.EnsureError(DocumentErrorSeverity.Severe, args[0], TexlStrings.ErrNeedValidVariableName_Arg, Name);
+                fValid = false;
+            }
+
+            // Get the unified collected type on the RHS. This will generate appropriate
+            // document errors for invalid arguments such as unsupported aggregate types.
+            fValid &= TryGetUnifiedCollectedType(args, argTypes, errors, out DType collectedType);
+            Contracts.Assert(collectedType.IsTable);
+
+            // The item type must be compatible with the collection schema.
+            var fError = false;
+            returnType = DType.Union(ref fError, collectionType, collectedType, useLegacyDateTimeAccepts: true);
+            if (fError)
+            {
+                fValid = false;
+                if (!SetErrorForMismatchedColumns(collectionType, collectedType, args[1], errors))
+                {
+                    errors.EnsureError(DocumentErrorSeverity.Severe, args[0], TexlStrings.ErrNeedValidVariableName_Arg);
+                }
+            }
+
+            return fValid;
+        }
+
+        // This method returns true if there are special suggestions for a particular parameter of the function.
+        public override bool HasSuggestionsForParam(int argumentIndex)
+        {
+            Contracts.Assert(argumentIndex >= 0);
+
+            return argumentIndex == 0;
+        }
+
+        public override bool IsAsyncInvocation(CallNode callNode, TexlBinding binding)
+        {
+            Contracts.AssertValue(callNode);
+            Contracts.AssertValue(binding);
+
+            return Arg0RequiresAsync(callNode, binding);
+        }
+
+        public async Task<FormulaValue> InvokeAsync(FormulaValue[] args, CancellationToken cancel)
+        {
+            var arg0 = (TableValue)args[0];
+            var arg1 = (RecordValue)args[1];
+
+            var mytype0 = arg0.GetType();
+            var mytype1 = arg1.GetType();
+
+            var result = arg0.AppendAsync(arg1);
+
+            return await Task.FromResult<FormulaValue>(FormulaValue.New(true));
```

Comment:
>Task.FromResult<FormulaValue> [](http://example.com/codeflow?start=25&length=29)

With await, This can just be:
return await FormulaValue.New(true);